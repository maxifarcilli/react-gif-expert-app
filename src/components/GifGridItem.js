import React from 'react';
import PropTypes from 'prop-types'; // snippet: impt.

const GifGridItem = ( { title , url } ) => {

    return (
        <div className='card animate__animated animate__fadeIn' >
            <img src={ url } alt={ title } />
            <p>{ title }</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}


export default GifGridItem;

/*
    Pruebas unitarias
    1
    ------------------------------------------------
        debe de mostrar el componente correctamente:
            - shallow
            - wrapper
            - wrapper.toMatchSnapshow()
    
    2
    ------------------------------------------------
        agregar proptypes requeridos en title y url.
*/