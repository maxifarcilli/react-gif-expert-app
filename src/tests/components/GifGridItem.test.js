import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas sobre el componente <GifGridItem />', () => {

    const title = 'Titulo';
    const url = 'https://localhost/nofind.jpg';
    const wrapper = shallow( 
        <GifGridItem 
            title = { title }
            url = { url }
        /> 
    );
    
    
    test('Debe de mostrar el componente correctamente ', () => {
         
        expect( wrapper ).toMatchSnapshot();

    });

    test('Debe de tener un pÃ¡rrafo con el title', () => {
       
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );

    });
    
    test('Debe de tener la imagen igual a url y alt de los props', () => {
       
        const img = wrapper.find('img');

        // Buscar properties en un elemento html con enzyme
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );

    });
    
    // Tarea resuelta por mi
    const className = 'animate__fadeIn';
    test('Debe de tener la clase css animate__fadeIn', () => {
        
        
        const div = wrapper.find('div');
        const existe = div.prop('className').includes(className);

        expect( existe ).toBe( true );

    });
    
    // Tarea resuelta por el curso
    test('Debe de tener la clase css animate__fadeIn (por el curso)', () => {
        
        
        const div = wrapper.find('div');
        const className = div.prop('className')

        expect( className.includes( 'animate__fadeIn' ) ).toBe( true );

    });

    
        
});
