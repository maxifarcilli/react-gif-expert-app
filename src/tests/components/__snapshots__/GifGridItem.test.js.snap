// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <GifGridItem /> Debe de mostrar el componente correctamente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo"
    src="https://localhost/nofind.jpg"
  />
  <p>
    Titulo
  </p>
</div>
`;
